doc-deploy-changelog/action.yml

```
@@ -73,21 +73,46 @@ inputs:
     default: true
     type: boolean
 
-  update-release-branch:
+  release-from-main:
     description: |
-      Whether or not to update the CHANGELOG in the release branch.
-      If ``true``, the branch used to update the CHANGELOG is "release/major.minor".
-      If ``false``, the branch used to update the CHANGELOG is
-      "maint/changelog-update-tag_version" and will be deleted after use.
-    required: false
-    default: true
-    type: boolean
+      There are two options when making a release for your repository:
 
-  update-main-branch:
-    description: |
-      Whether or not to update the CHANGELOG in the main branch.
-      If ``true``, the branch used to update the CHANGELOG is main.
-      If ``false``, check the value of the update-release-branch input.
+      1. You pushed a tag from a release branch
+
+        This is the most common case for PyAnsys repositories. The following steps are
+        performed when ``release-from-main`` is ``false``:
+
+        a) Find a release branch that has the same major and minor version of the
+        tag that was pushed. For example, if your tag was "v0.1.2", it looks for
+        a release branch named "release/0.1". If the release branch cannot be found, a
+        temporary branch is used named "maint/changelog-update-vTAG_VERSION".
+
+        b) Checkout the release or temporary branch, update the CHANGELOG file
+        using ``towncrier``, and push the updates to the respective branch.
+
+        c) Checkout the main branch and create a pull request branch named
+        "maint/vTAG_VERSION-changelog". Delete the pull request branch from the remote
+        if it exists, checkout the pull request branch, cherry pick the commit
+        containing the CHANGELOG file changes from the previous step, push the
+        pull request branch, and create a pull request from the branch into main.
+
+        d) Checkout the release or temporary branch, delete the tag locally and
+        on the remote, create the new tag locally, and push the tag on the remote.
+        Delete the temporary branch if one was used to update the CHANGELOG file.
+
+        e) Exit on error if the CHANGELOG is updated.
```

Might be worth saying something here about why it exits on error:

```suggestion
        e) Exit on error if the CHANGELOG is updated, so that the package is not released from **this** instance of the workflow
```

---

Or maybe clarifying somewhere that with this action, pushing a release tag results in the entire workflow running twice:

* Once to run towncrier and consolidate the changelog fragments into a single update of the CHANGELOG
* Once to release the package after the CHANGELOG has been updated

It makes sense once you see everything work, but that first time is always a bit nerve-wracking when you see it fail!

---

Sure, I can add that in!

---

Thanks!